### ISSUE TITLE
feature importance

### ISSUE THREAD

**Issue by MichaelMedek** [dholzmueller/pytabkit#13]
For standard XGBoost you can use

```python
model.get_score(importance_type='gain')

>>{'ftr_col1': 77.21064539577829,
   'ftr_col2': 10.28690566363971,
   'ftr_col3': 24.225014841466294,
   'ftr_col4': 11.234086283060112}
```

To get feature importance. Is it planned to have this also in your lib? Would be really helpful!

**Comment by dholzmueller** [comment]
Hi, it is currently not planned. Would you be specifically interested in the exact same methodology for the XGBoost wrappers in the lib, or some more general functionality for all models? In the latter case, you could just use permutation importance from scikit-learn, which works for any scikit-learn classifier or regressor: https://scikit-learn.org/stable/modules/permutation_importance.html

**Comment by MichaelMedek** [comment]
Thanks for your quick answer! I have chosen your lib for the great and robust pre-processing code and the nice selection of model with great defaults chosen.
My input are pandas data frame with all kind of messy column types. Your pre-processing code is handling it nicely and your models giving me great accuracy, while vanilla XGBoost would need additional pre-processing. 
Would this proposes scikit learn feature importance be independent of the pre-processing?

**Comment by dholzmueller** [comment]
The scikit-learn feature importance checks how much the loss increases if the values of a feature are permuted. So it is usually independent of the preprocessing.
By the way, if you just want maximum accuracy without manual preprocessing, you could check out AutoGluon, which does model tuning/ensembling of many models with a given time budget. It doesn't have all the latest models but much more sophisticated model combination logic so there is a good chance it will perform even better.
